### PR TITLE
Yaml generator environment variable check

### PR DESCRIPTION
**Pull request by bhilbert4**
This PR pushes the MIRAGE_DATA environment variable expansion function into `utils.py`, as it is called in multiple places within Mirage. It also adds a check that the directory specified by the environment variable exists. It was previously checking that the environment variable was set, but in the case where the specified directory didn't exist (or wasn't accessible due to forgetting to log in to the VPN), the resulting exception wasn't very helpful. This PR changes that so that a more informative error statement is given.

### CODE REVIEW

**Comment by bhilbert4** [comment]
This addresses #255

**Comment by bhilbert4** [comment]
Pretty simple change to the code. @Johannes-Sahlmann, do you have a moment to look over the changes and make sure they make sense? I ran the `yaml_generator` and created a simulated ramp using the changes and everything worked fine.

**Comment by Johannes-Sahlmann** [comment]
I will have a look asap.

File: mirage/utils/utils.py [hunk 1]
```diff
@@ -157,6 +157,34 @@ def ensure_dir_exists(fullpath):
...
+        Environment variable name
+
+    Returns
+    -------
+    variable_directory : str
+        Path pointed to by the environment variable
+    """
+    variable_directory = os.environ.get(variable_name)
+    if variable_directory is None:
+        raise ValueError(("WARNING: {} environment variable is not set. "
```
Comment (Johannes-Sahlmann) [inline mirage/utils/utils.py]: I suggest to remove the `WARNING:` part of the message.

File: mirage/utils/utils.py [hunk 2]
```diff
@@ -157,6 +157,34 @@ def ensure_dir_exists(fullpath):
...
+                          "This must be set to the base directory "
+                          "containing the darks, cosmic ray, PSF, etc "
+                          "input files needed for the simulation. "
+                          "These files must be downloaded separately "
+                          "from the Mirage package.".format(variable_name)))
+    if not os.path.isdir(variable_directory):
+        raise FileNotFoundError(("WARNING: The directory contained in the {} "
```
Comment (Johannes-Sahlmann) [inline mirage/utils/utils.py]: Same here. I suggest to remove the warning part.

File: mirage/utils/utils.py (hunk 1 above)
Comment (bhilbert4) [reply mirage/utils/utils.py]: Removed.

File: mirage/utils/utils.py (hunk 2 above)
Comment (bhilbert4) [reply mirage/utils/utils.py]: Removed

**Comment by bhilbert4** [comment]
Arg. These changes cause Travis tests to fail because it now checks for the existence of the MIRAGE_DATA directory. I'll have to add an `offline` flag similar to what we have for the `yaml_generator` I guess.

**Comment by Johannes-Sahlmann** [comment]
I should have waited for the checks to finish before approving ...

**Comment by bhilbert4** [comment]
All tests now passing locally and on Travis!! I had to add 'offline' keywords to all the main files: **catalog_seed_image.py** **dark_prep.py**, **obs_generator.py**, **imaging_simulator.py** and **wfss_simulator.py**.

**Pull request merged**